kubernetes/samples/scripts/create-weblogic-domain/domain-home-on-pv/create-weblogic-sample-domain.sh

```diff
@@ -348,16 +365,66 @@ function createYamlFiles {
   sed -i -e "s:%ADMIN_NODE_PORT%:${adminNodePort}:g" ${dcrOutput}
   sed -i -e "s:%JAVA_OPTIONS%:${javaOptions}:g" ${dcrOutput}
   sed -i -e "s:%STARTUP_CONTROL%:${startupControl}:g" ${dcrOutput}
-
+ 
   # Remove any "...yaml-e" files left over from running sed
   rm -f ${domainOutputDir}/*.yaml-e
 }
 
+# create domain configmap using what is in the createDomainFilesDir
+function create_domain_configmap {
+  # Use the default files if createDomainFilesDir is not specified
+  if [ -z "${createDomainFilesDir}" ]; then
+    createDomainFilesDir=${scriptDir}/wlst
+  fi
+
+  # customize the files with domain information
+  local externalFilesTmpDir=$domainOutputDir/tmp
+  mkdir -p $externalFilesTmpDir
+  cp ${createDomainFilesDir}/* ${externalFilesTmpDir}/
+ 
+  for fname in ${externalFilesTmpDir}/*
+  do
+    local outputFile=$fname
+    sed -i -e "s:%NAMESPACE%:$namespace:g" ${outputFile}
+    sed -i -e "s:%WEBLOGIC_CREDENTIALS_SECRET_NAME%:${weblogicCredentialsSecretName}:g" ${outputFile}
+    sed -i -e "s:%WEBLOGIC_IMAGE%:${weblogicImage}:g" ${outputFile}
+    sed -i -e "s:%WEBLOGIC_IMAGE_PULL_SECRET_NAME%:${weblogicImagePullSecretName}:g" ${outputFile}
+    sed -i -e "s:%WEBLOGIC_IMAGE_PULL_SECRET_PREFIX%:${weblogicImagePullSecretPrefix}:g" ${outputFile}
+    sed -i -e "s:%DOMAIN_UID%:${domainUID}:g" ${outputFile}
+    sed -i -e "s:%DOMAIN_NAME%:${domainName}:g" ${outputFile}
+    sed -i -e "s:%PRODUCTION_MODE_ENABLED%:${productionModeEnabled}:g" ${outputFile}
+    sed -i -e "s:%ADMIN_SERVER_NAME%:${adminServerName}:g" ${outputFile}
+    sed -i -e "s:%ADMIN_SERVER_NAME_SVC%:${adminServerNameSVC}:g" ${outputFile}
+    sed -i -e "s:%ADMIN_PORT%:${adminPort}:g" ${outputFile}
+    sed -i -e "s:%CONFIGURED_MANAGED_SERVER_COUNT%:${configuredManagedServerCount}:g" ${outputFile}
+    sed -i -e "s:%MANAGED_SERVER_NAME_BASE%:${managedServerNameBase}:g" ${outputFile}
+    sed -i -e "s:%MANAGED_SERVER_NAME_BASE_SVC%:${managedServerNameBaseSVC}:g" ${outputFile}
+    sed -i -e "s:%MANAGED_SERVER_PORT%:${managedServerPort}:g" ${outputFile}
+    sed -i -e "s:%T3_CHANNEL_PORT%:${t3ChannelPort}:g" ${outputFile}
+    sed -i -e "s:%T3_PUBLIC_ADDRESS%:${t3PublicAddress}:g" ${outputFile}
+    sed -i -e "s:%CLUSTER_NAME%:${clusterName}:g" ${outputFile}
+    sed -i -e "s:%CLUSTER_TYPE%:${clusterType}:g" ${outputFile}
+    sed -i -e "s:%DOMAIN_PVC_NAME%:${persistentVolumeClaimName}:g" ${outputFile}
+    sed -i -e "s:%DOMAIN_ROOT_DIR%:${domainPVMountPath}:g" ${outputFile}
+    sed -i -e "s:%DOMAIN_LOGS_DIR%:${domainPVMountPath}/logs:g" ${outputFile}
+    sed -i -e "s:%CREATE_DOMAIN_SCRIPT_DIR%:${createDomainScriptsMountPath}:g" ${outputFile}
```

Comment:
Do you really want different vars for the create domain script directory and name, instead of just a single pathname?

Reply:
There can be multiple shell scripts involved in creating the domain home. We need to know the script that the job needs to run.

Reply:
ok